ingest/validation/validate_metadata.py

```diff
@@ -1402,6 +1410,81 @@ def detect_excel_drag(metadata, convention):
     return excel_drag
 
 
+def to_1D(series):
+    """ Pandas values which are list need unnesting
+    """
+    return [x for _list in series for x in _list]
+
+
+def replace_single_value_array(df, metadata_name, synonym, label):
+    """ Synonym replacement (in-place) for single-value array metadata
+    Pandas doesn't operate well on lists which are potentially non-homogenous
+    # https://stackoverflow.com/questions/53116286/how-to-assign-an-entire-list-to-each-row-of-a-pandas-dataframe
+    """
+    match = [v == [synonym] for v in df[metadata_name]]
+    value = [label]
+    df.loc[match, metadata_name] = df.apply(lambda x: value, axis=1)
+
+
+def replace_synonym_in_multivalue_array(df, metadata_name, substitutions):
+    """ Synonym replacement (in-place) for multi-value array ontology labels
+        must identify all affected arrays of labels, construct replacement arrays
+        then replace old synonym-containing array with an updated array
+    """
+    orig_values = list(df[metadata_name].transform(tuple).unique())
+    matching_synonyms = {}
+    for o in orig_values:
+        # if a synonym (s) is an element of the multivalue array (o), track it
+        matching_synonyms[o] = [s for s in substitutions.keys() if s in o]
+    for o in matching_synonyms.keys():
+        # if a multivalue array contains synonyms
+        if matching_synonyms[o]:
+            # make a copy of the original multivalue array that will take on all substitutions
+            replacement_value = list(o)
+            # make all synonym substitutions into the multivalue array of ontology labels
+            for s in matching_synonyms[o]:
+                replacement_value = [
+                    substitutions[s] if term == s else term
+                    for term in replacement_value
+                ]
+            # select the rows in the dataframe with entries for multivalue array (o)
+            match = [v == list(o) for v in df[metadata_name]]
+            df.loc[match, metadata_name] = df.apply(lambda x: replacement_value, axis=1)
+
+
+def replace_synonyms(metadata):
+    """
+    Update BigQuery data to store ontology labels and not synonyms
+    """
+    bq_filename = str(metadata.study_file_id) + ".json"
```

Comment:
Study accession + file name seems easier to reason about than study file ID.

```suggestion
    bq_filename = str(metadata.study_file_id) + ".json"
```

Reply:
I agree a study accession is more readable. However, when in "developer mode" we bypass MongoDB lookups and only have the same hard-coded dummy accession "SCPdev" regardless of what is being ingested. Developers are able to pass in variable (BSON compliant) dummy study_file_id so, for developer productivity reasons, using the study_file_id is preferable.